### PR TITLE
Image modifier changes

### PR DESCRIPTION
**Pull request by shahi645**
- Reformat image to make answers more prominent

### CODE REVIEW

File: transform/transformers/PDFTransformer.py [hunk 1]
```diff
@@ -7,12 +7,14 @@
 import os
 import sys
 import uuid
+from copy import copy
```
Comment (JamesGardiner) [inline transform/transformers/PDFTransformer.py]: Try and keep imports alphabetised

File: transform/transformers/PDFTransformer.py [hunk 2]
```diff
@@ -28,9 +30,17 @@
 styles = getSampleStyleSheet()
 
 # Basic text style
-styleN = styles["BodyText"]
+styleN = copy(styles["BodyText"])
```
Comment (JamesGardiner) [inline transform/transformers/PDFTransformer.py]: Are you aware that this is only returning a shallow copy of `styles["BodyText"]` the difference being:


    - A shallow copy constructs a new compound object and then (to the extent possible) inserts references into it to the objects found in the original.
    - A deep copy constructs a new compound object and then, recursively, inserts copies into it of the objects found in the original.

You might want to consider which to use here (though that's not to say this is the wrong choice!)

File: transform/transformers/PDFTransformer.py [hunk 3]
```diff
@@ -28,9 +30,17 @@
...
 styleN.alignment = TA_LEFT
-styleN.spaceAfter = 25
+
+# Answer Style
+styleAnswer = copy(styles["BodyText"])
```
Comment (JamesGardiner) [inline transform/transformers/PDFTransformer.py]: can we change var names to be PEP8 compliant https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables `lowercase with words separated by underscores as necessary to improve readability`

File: .idea/vcs.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (JamesGardiner) [inline .idea/vcs.xml]: This shouldn't have been commited. Remove this from the git repo, and add .idea to your global .gitignore

File: tests/test_cora_transform.py [hunk 5]
```diff
@@ -413,8 +413,8 @@ def test_ukis_pdf(self):
         pages = list(PackerTests.extract_text(path))
         self.assertEqual(2, len(pages))
         questions = [
-            ("3.3", "3.5", "3.8", "3.10", "3.12", "3.14", "3.17"),
-            ("11.1", "11.2", "11.3", "13.1")
+            ("3.3", "3.5", "3.8", "3.10", "3.12", "3.14", "3.17", "11.1"),
```
Comment (JamesGardiner) [inline tests/test_cora_transform.py:416]: Why is 11.1 now in the first tuple?

File: transform/transformers/PDFTransformer.py [hunk 6]
```diff
@@ -107,18 +117,30 @@ def get_elements(self):
         elements.append(heading)
 
         for question_group in self.survey['question_groups']:
+            has_section_heading_output = False
 
             if 'title' in question_group:
-                elements.append(Paragraph(question_group['title'], styleSH))
 
                 for question in question_group['questions']:
```
Comment (JamesGardiner) [inline transform/transformers/PDFTransformer.py]: Could be rewritten something like this:

```python
for question_group in filter(lambda x: 'title' in x, self.survey['question_groups']):

            has_section_heading_output = False

            for question in filter(lambda x: 'text' in x, question_group['questions']):
```

File: transform/transformers/PDFTransformer.py [hunk 7]
```diff
@@ -107,18 +117,30 @@ def get_elements(self):
         elements.append(heading)
 
         for question_group in self.survey['question_groups']:
```
Comment (JamesGardiner) [inline transform/transformers/PDFTransformer.py]: Could be rewritten something like this:

```python
for question_group in filter(lambda x: 'title' in x, self.survey['question_groups']):
    has_section_heading_output = False
    for question in filter(lambda x: 'text' in x, question_group['questions']):
```

File: transform/transformers/PDFTransformer.py [hunk 8]
```diff
@@ -107,18 +117,30 @@ def get_elements(self):
...
+            has_section_heading_output = False
 
             if 'title' in question_group:
-                elements.append(Paragraph(question_group['title'], styleSH))
 
                 for question in question_group['questions']:
                     if 'text' in question:
                         answer = ''
                         if question['question_id'] in self.response['data']:
```
Comment (JamesGardiner) [inline transform/transformers/PDFTransformer.py]: This  would be an ideal try/except block

```python
try:
    answer = self.response['data'][question['question_id']]
except KeyError:
    answer = ''
```

File: transform/transformers/PDFTransformer.py [hunk 9]
```diff
@@ -107,18 +117,30 @@ def get_elements(self):
...
+                            # Output the section header if we haven't already
+                            # Checking here so that whole sections are suppressed
+                            # if they have no answers.
+                            if not has_section_heading_output:
+                                elements.append(HRFlowable(width="100%"))
+                                elements.append(Paragraph(question_group['title'], styleSH))
+                                has_section_heading_output = True
+
+                            # Question not output if answer is empty
+                            text = question.get("text", " ")
```
Comment (JamesGardiner) [inline transform/transformers/PDFTransformer.py]: didn't you check above that text is present in question?

File: transform/transformers/PDFTransformer.py [hunk 10]
```diff
@@ -141,11 +163,11 @@ def main(args):
     with open(fP, "r") as fObj:
```
Comment (JamesGardiner) [inline transform/transformers/PDFTransformer.py]: pep8 variable name again

File: transform/transformers/PDFTransformer.py [hunk 11]
```diff
@@ -141,11 +163,11 @@ def main(args):
...
 
-    data = json.load(sys.stdin)
-    tx = PDFTransformer(survey, data)
-    output = tx.render()
-    sys.stdout.write(output.decode("latin-1"))
-    return 0
+        data = json.load(sys.stdin)
+        tx = PDFTransformer(survey, data)
+        output = tx.render()
+        sys.stdout.write(output.decode("latin-1"))
```
Comment (JamesGardiner) [inline transform/transformers/PDFTransformer.py]: Is latin-1 a requirement of the downstream??

**Review by JamesGardiner — CHANGES_REQUESTED** [review]

File: transform/transformers/PDFTransformer.py (hunk 1 above)
Comment (shahi645) [reply transform/transformers/PDFTransformer.py]: done

File: transform/transformers/PDFTransformer.py (hunk 2 above)
Comment (shahi645) [reply transform/transformers/PDFTransformer.py]: done

File: transform/transformers/PDFTransformer.py (hunk 3 above)
Comment (shahi645) [reply transform/transformers/PDFTransformer.py]: done

File: transform/transformers/PDFTransformer.py (hunk 7 above)
Comment (shahi645) [reply transform/transformers/PDFTransformer.py]: done

File: transform/transformers/PDFTransformer.py (hunk 6 above)
Comment (shahi645) [reply transform/transformers/PDFTransformer.py]: done

File: transform/transformers/PDFTransformer.py (hunk 8 above)
Comment (shahi645) [reply transform/transformers/PDFTransformer.py]: done

File: transform/transformers/PDFTransformer.py (hunk 9 above)
Comment (shahi645) [reply transform/transformers/PDFTransformer.py]: done

File: transform/transformers/PDFTransformer.py (hunk 10 above)
Comment (shahi645) [reply transform/transformers/PDFTransformer.py]: done

File: transform/transformers/PDFTransformer.py (hunk 11 above)
Comment (shahi645) [reply transform/transformers/PDFTransformer.py]: changed to utf-8

**Review by JamesGardiner — APPROVED** [review]

File: transform/transformers/PDFTransformer.py [hunk 12]
```diff
@@ -137,15 +162,15 @@ def parser(description=__doc__):
...
-    sys.stdout.write(output.decode("latin-1"))
-    return 0
+    fp = os.path.expanduser(os.path.abspath(args.survey))
+    with open(fp, "r") as f_obj:
+        survey = json.load(f_obj)
+
+        data = json.load(sys.stdin)
+        tx = PDFTransformer(survey, data)
+        output = tx.render()
+        sys.stdout.write(output.decode("utf-8"))
```
Comment (tundish) [inline transform/transformers/PDFTransformer.py]: _latin-1_  was the necessary encoding.

**Review by tundish — CHANGES_REQUESTED** [review]

**Review by tundish — APPROVED** [review]

**Pull request merged**